script.js

```diff
@@ -0,0 +1,91 @@
+const X_CLASS = 'playerX'
+const O_CLASS = 'playerO'
+const winningCondition = [
+[0, 1, 2],
+[3, 4, 5],
+[6, 7, 8],
+[0, 3, 6],
+[1, 4, 7],
+[2, 5, 8],
+[0, 4, 8],
+[2, 4, 6]
+];
+
+const cellElements = document.querySelectorAll('.cell')
+const board = document.getElementById('grid')
+const winningMessageElement = document.getElementById('winningMessage')
+const restartButton = document.getElementById('restartButton')
+const winningMessageTextElement = document.querySelector('.data-winning-message-text')
+let circleTurn
+
+startGame()
+
+restartButton.addEventListener('click', startGame)
+
+function startGame() {
+  circleTurn = false
+  cellElements.forEach(cell => {
+    cell.classList.remove(X_CLASS)
+    cell.classList.remove(O_CLASS)
+    cell.removeEventListener('click', handleClick)
+    cell.addEventListener('click', handleClick, { once: true })
+  })
+  setBoardHoverClass()
+  winningMessageElement.classList.remove('show')
+}
+
+function handleClick(e) {
+    const cell = e.target
+    console.log(cell)
+    const currentClass = circleTurn ? O_CLASS : X_CLASS
+    placeMark(cell, currentClass)
+    if (checkWin(currentClass)) {
+      endGame(false)
+    } else if (isDraw()) {
+      endGame(true)
+    } else {
+      swapTurns()
+      setBoardHoverClass()
+    }
+  }
+
+  function endGame(draw) {
+    if (draw) {
+      winningMessageTextElement.innerText = 'Draw!'
+    } else {
+      winningMessageTextElement.innerText = `${circleTurn ? "O's" : "X's"} Wins!`
+    }
+    winningMessageElement.classList.add('show')
+  }
+
+  function isDraw() {
+    return [...cellElements].every(cell => {
```

Comment:
this is some advanced stuff! what is every doing here? and the [...]?

Reply:
hello Pete, like I said when we presented the homework in class, I got most of these codes from the internet, I literally frankensteined some lines until I got it to do what I wanted.. from what I understand after looking it up, the three dots are called a spread operator and it calls everything available from, in this case, the "grid", which is every div that has the ID ".cell" and treats it as an array.

for instance if I had:

var A = [1, 2, 3, 4]
var B = [...A, 4, 5, 6]
var C = [7, 8, ...A]

if I console.log(B) - it should give the array [1, 2, 3, 4, 5, 6]

if I console.log(C) - it should give the array [7, 8, 1, 2, 3, 4]


thank you for the feedback

Reply:
totally correct understanding! But in this case the spread isn't being used to add elements into an array. I realized after I wrote this that what it's doing is converting the NodeList that results from querySelectorAll into an array.